sympy/solvers/solveset.py

```diff
@@ -330,6 +330,15 @@ def _invert_complex(f, g_ys, symbol):
                 return (h, S.EmptySet)
             return _invert_complex(h, imageset(Lambda(n, n/g), g_ys), symbol)
 
+    if f.is_Pow:
+        base, expo = f.args
+        expo_has_sym = expo.has(symbol)
+
+        if not expo_has_sym:
+            if expo.is_rational and g_ys == FiniteSet(0):
```

Comment:
I think this could be simplified by checking `if expo.is_Rational` which checks if it is a Rational rather than e.g. `Symbol('x', rational=True)`.

Reply:
Right. Didn't notice the difference between `is_rational` and `is_Rational`.